src/Inputs/SearchBar.tsx

```diff
@@ -0,0 +1,236 @@
+    value?: string | number;
+    onChange?: Function;
+    limit?: number;
+    numVisible: number;
+}
+
+export const SearchBar: React.FC<SearchBarProps> = ({
+    suggestiveOptions,
+    value,
+    limit = 4,
+    onChange = (): void => {},
+    name,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const [inputValue, setinputValue] = useState('');
+    const [, mount, animation] = useTransition(expanded);
+    const refSelectList = useRef() as RefObject<HTMLDivElement>;
+    const [numVisibleSelection, setNumVisibleSelection] = useState(limit);
+
+    const createList = (
+        children: string[],
+        onSelect: React.MouseEventHandler,
+    ): React.ReactNode[] =>
+        children.map((child): React.ReactElement | null => {
+            if (!inputValue) {
+                if (child) {
+                    const val = child;
+                    const selected = String(value) === val;
+                    return (
+                        <SelectItem
+                            active={selected}
+                            onClick={onSelect}
+                            key={val as string}
+                            order="1"
+                        >
+                            {child}
+                        </SelectItem>
+                    );
+                }
+            } else if (
+                child &&
+                inputValue &&
+                child.toLowerCase().search(inputValue.toLowerCase()) > -1
+            ) {
+                const val = child;
+                const selected = String(value) === val;
+                const order = child
+                    .toLowerCase()
+                    .startsWith(inputValue.toLowerCase())
+                    ? '-1'
+                    : '1';
+                return (
+                    <SelectItem
+                        {...child}
+                        active={selected}
+                        onClick={onSelect}
+                        key={val as string}
+                        order={order as string}
+                    >
+                        {child}
+                    </SelectItem>
+                );
+            }
+
+            return null;
+        });
+
+    const onSelect = useCallback(
+        ({ currentTarget }): void => {
+            onChange({
+                target: {
+                    value: currentTarget.getAttribute('value'),
+                    name,
+                },
+            });
+            setinputValue(currentTarget.firstChild.nodeValue);
+        },
+        [name],
+    );
+
+    const options = useMemo((): any[] => {
+        return Children.toArray(suggestiveOptions);
+    }, [expanded]);
+
+    useLayoutEffect((): void | (() => void) => {
+        if (refSelectList.current?.children.length) {
+            setNumVisibleSelection(refSelectList.current?.children.length);
+        }
+
+        const listener = (): void => {
```

Comment:
Shouldn't this be deleted?

Reply:
@ralph-dev  but then where should I setExpanded to false when clicks or keydown events happen?

Reply:
Ok you are right, I thought this was a generic listeners becuase of the name (Maybe name is ExpandedListener instead?)